### PR TITLE
Make e2e test repairs more robust

### PR DESCRIPTION
**Pull request by adrianludwin**
Made several debuggability and functional improvements:

* Added timestamps to all output to correlate with logs. Timestamps are
  of the form "seconds since Unix epoch" which isn't human friendly but
  is identical to the timestamps produced by HNC logs.
* Fully deleted the HNC deployment in RecoverHNC since HNC doesn't
  really seem to recover well if various things are changed without
  restarting the pod.
* Make the post-recovery test far more robust by not ignoring failures
  to delete the test namespaces.
* Stop force-deleting pods in the rolebinding test and instead just wait
  a few moments. I found that we don't actually need to wait for the pod
  object to be fully deleted on the server for it to stop getting in the
  way; the container in the pod appears to stop running ~instantly while
  the pod can occasionally hang around for over 60s.

All of these were added as I saw failures in the affected code.

Tested: Ran a set of five flaky tests with and without these changes
(while also including PR #1163). Without, at least one of them failed
virtually every time; with this change, they passed 5/5 times on GKE
plus 2/2 times on Kind.

### CODE REVIEW

File: incubator/hnc/pkg/testutils/testutils.go [hunk 1]
```diff
@@ -257,13 +263,34 @@ func RecoverHNC() {
...
+	// TODO: if CleanupNamespaces ever starts using labels to select namespaces to delete, then get
+	// rid of this hack.
+	if err := TryRunQuietly("kubectl get ns", a); err == nil {
+		MustRunWithTimeout(30, "kubectl hns set", a, "-a")
+		MustRunWithTimeout(30, "kubectl delete ns", a)
+	}
+	if err := TryRunQuietly("kubectl get ns", b); err == nil {
+		MustRunWithTimeout(30, "kubectl annotate ns", b, "hnc.x-k8s.io/subnamespaceOf-")
+		MustRunWithTimeout(30, "kubectl delete ns", b)
+	}
```
Comment (yiqigao217) [inline incubator/hnc/pkg/testutils/testutils.go:285]: Hmm, this looked the same to me as `CleanupNamespaces()` except skipping de-annotating a and no setting AC for b. What makes a difference?

Comment (adrianludwin) [reply incubator/hnc/pkg/testutils/testutils.go:285]: CleanupNamespaces is TryRunQuietly, this is MustRunWithTimeout.

**Review by yiqigao217 — COMMENTED** [review]
Ah I see.
/lgtm
/hold
/assign @rjbez17

**Comment by adrianludwin** [comment]
Test-only changes don't need double-approval, though @rjbez17 please lmk if you have any suggestions and I'll follow them up.

/hold cancel

**Pull request merged**